src/SourceNodes/hlsnode.ts

```diff
@@ -83,6 +85,21 @@ class HLSNode extends MediaNode {
         super._load();
     }
 
+    _isReady() {
+        if (this._state === SOURCENODESTATE.sequenced || this._state === SOURCENODESTATE.waiting) {
```

Comment:
we sill need to return `false` if the current node is the one playing and is buffering

Reply:
`* 0 - Waiting, start() has not been called on it yet.
* 1 - Sequenced, start() has been called but it is not playing yet.`

Waiting basically means the source node has been created but it hasn't been given a start time.

I think isReady effectively is the same as checking _buffered for our HLSNode, but I'll double check. It might be enough to have this waiting/sequenced check and then call super.

Reply:
Ok, I figured out what was actually causing the pauses.. it wasn't future source nodes that were pausing it, it was *past* nodes that were still buffering... Since I've made the HLSNode stop loading once _unload is called, these nodes will permanently stall playback --- I think this code is now equivalent to your previous code. 

There's still an issue of why the playback didn't stall when the node that was buffering didn't finish buffering.. and how it was skipped over... but this could be due to all of the debug rendering we're now doing and possible the RAF took so long that it had skipped onto the next node and unloaded the previous one before it could stall -- I also only saw this at 8x playback speed.

Reply:
technically with this code is possible that `_isReady` fn returns `true` for a node in `waiting` state and currently _playing_, in this edge case the player will try to play without content
will this work fine if we change it to the following code?:

```
_isReady() {
    if (this._state === STATE.playing) {
        return super._isReady();
    } else {
        if (this._state === SOURCENODESTATE.sequenced || this._state === SOURCENODESTATE.waiting) {
            return true;
        }
        return this._ready;
    }
}
```

Reply:
A SourceNode can't be in state `waiting` AND `playing` -- if it is that'd be a bug.

A SourceNode can only get into a playing state, from either
- the PAUSED state (via a call to play). [See Here](https://github.com/alanjrogers/VideoContext/blob/890c9435abc476fb413ed62fa0f637edee16ea5c/src/SourceNodes/sourcenode.ts#L357)
OR 
- via _update when the current time is passed the node's startTime, a SourceNode in a waiting state doesn't have a startTime and so will never enter a playing state (_startTime is set to NaN when a node is in waiting state) [See here](https://github.com/alanjrogers/VideoContext/blob/890c9435abc476fb413ed62fa0f637edee16ea5c/src/SourceNodes/sourcenode.ts#L405)

Reply:
> Waiting basically means the source node has been created but it hasn't been given a start time.

Another way to say this is that it's not on yet on the timeline.. it needs at least a start time, and then it'll be sequenced and get into a buffering state when it hits it's preload window.

Reply:
👍 ok sounds good then